gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -591,6 +592,33 @@ public static QueryResultsOption maxWaitTime(long maxWaitTime) {
   InsertAllResponse insertAll(InsertAllRequest request) throws BigQueryException;
 
   /**
+   * Sends a resumable insert request given a seekable channel containing the rows to be inserted.
+   * This method does not close the channel so you should take care of closing it.
+   *
+   * <p>Example usage of inserting data from a local file:
+   * <pre> {@code
+   * LoadConfiguration config = LoadConfiguration.of(TableId.of("my_dataset_id", "my_table_id"));
+   * try(FileChannel channel = FileChannel.open(Paths.get("/path/to/file"))) {
+   *   bigquery.insertAll(config, channel);
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @see <a href="https://cloud.google.com/bigquery/loading-data-post-request#resumable">Resumable
+   *     Upload</a>
+   */
+  void insertAll(LoadConfiguration configuration, SeekableByteChannel channel)
```

Comment:
Yes I oversimplified the behavior of BufferedInputStream :) What I intended to say is that in the current implementation we cannot afford to use it.

I will implement the `WriteChannel` approach, we can always add this direct method later if need be.

I think the `WriteChannel` is good enough